x/ecocredit/server/core/msg_create_project_test.go

```diff
@@ -30,72 +30,120 @@ func (s *createProjectSuite) Before(t gocuke.TestingT) {
 
-	fee := sdk.Coin{
-		Denom:  "regen",
-		Amount: sdk.NewInt(20),
-	}
+	cKey, err := s.k.stateStore.ClassTable().InsertReturningID(s.ctx, &api.Class{
+		Id:               a,
+		CreditTypeAbbrev: creditTypeAbbrev,
+	})
+	require.NoError(s.t, err)
 
-	allowListEnabled := false
-	utils.ExpectParamGet(&allowListEnabled, s.paramsKeeper, core.KeyAllowlistEnabled, 1)
-	coinFee := sdk.Coins{fee}
-	utils.ExpectParamGet(&coinFee, s.paramsKeeper, core.KeyCreditClassFee, 1)
+	err = s.k.stateStore.ClassIssuerTable().Insert(s.ctx, &api.ClassIssuer{
+		ClassKey: cKey,
+		Issuer:   s.alice,
+	})
+	require.NoError(s.t, err)
+}
 
-	s.bankKeeper.EXPECT().SendCoinsFromAccountToModule(gmAny, gmAny, gmAny, gmAny).Return(nil).AnyTimes()
+func (s *createProjectSuite) AProjectSequenceForCreditClass(a, b string) {
+	class, err := s.k.stateStore.ClassTable().GetById(s.ctx, b)
+	require.NoError(s.t, err)
 
-	s.bankKeeper.EXPECT().BurnCoins(gmAny, gmAny, gmAny).Return(nil).AnyTimes()
+	nextSequence, err := strconv.ParseUint(a, 10, 32)
+	require.NoError(s.t, err)
 
-	_, err := s.k.CreateClass(s.ctx, &core.MsgCreateClass{
-		Admin:            s.alice.String(),
-		Issuers:          []string{s.alice.String()},
-		CreditTypeAbbrev: a,
-		Fee:              &fee,
+	err = s.k.stateStore.ProjectSequenceTable().Save(s.ctx, &api.ProjectSequence{
+		ClassKey:     class.Key,
+		NextSequence: nextSequence,
 	})
 	require.NoError(s.t, err)
 }
 
-func (s *createProjectSuite) AliceHasCreatedAProjectWithCreditClassId(a string) {
-	_, s.err = s.k.CreateProject(s.ctx, &core.MsgCreateProject{
-		Admin:        s.alice.String(),
-		ClassId:      a,
-		Jurisdiction: "US",
+func (s *createProjectSuite) AProjectWithProjectId(a string) {
+	classId := core.GetClassIdFromProjectId(a)
+
+	class, err := s.k.stateStore.ClassTable().GetById(s.ctx, classId)
+	require.NoError(s.t, err)
+
+	err = s.k.stateStore.ProjectTable().Insert(s.ctx, &api.Project{
+		Id:       a,
+		ClassKey: class.Key,
+	})
+	require.NoError(s.t, err)
+
+	seq := s.getProjectSequence(a)
+
+	err = s.k.stateStore.ProjectSequenceTable().Save(s.ctx, &api.ProjectSequence{
+		ClassKey:     class.Key,
+		NextSequence: seq + 1,
 	})
+	require.NoError(s.t, err)
 }
 
-func (s *createProjectSuite) TheProjectSequenceForCreditClassIs(a, b string) {
-	class, err := s.k.stateStore.ClassTable().GetById(s.ctx, a)
+func (s *createProjectSuite) AProjectWithProjectIdAndReferenceId(a, b string) {
+	classId := core.GetClassIdFromProjectId(a)
+
+	class, err := s.k.stateStore.ClassTable().GetById(s.ctx, classId)
 	require.NoError(s.t, err)
 
-	nextSequence, err := strconv.ParseUint(b, 10, 32)
+	err = s.k.stateStore.ProjectTable().Insert(s.ctx, &api.Project{
+		Id:          a,
+		ClassKey:    class.Key,
+		ReferenceId: b,
+	})
 	require.NoError(s.t, err)
 
-	err = s.k.stateStore.ProjectSequenceTable().Insert(s.ctx, &api.ProjectSequence{
+	seq := s.getProjectSequence(a)
+
+	err = s.k.stateStore.ProjectSequenceTable().Save(s.ctx, &api.ProjectSequence{
```

Comment:
```suggestion
	// Save because project sequence may already exist
	err = s.k.stateStore.ProjectSequenceTable().Save(s.ctx, &api.ProjectSequence{
```